src/main/java/dukechatbot/utility/TaskList.java

```diff
@@ -0,0 +1,132 @@
+package dukechatbot.utility;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * The TaskList implements the operations and attributes needed to handle
+ * manipulation of the task list.
+ */
+public class TaskList {
+    /**
+     * Encapsulates the array list associated with the instance of
+     * TaskList
+     */
+    private ArrayList<Task> tl;
+    private Ui ui;
+
+    /**
+     * Constructs the instance of Task List with its associated
+     * array list.
+     * @param tl the array list that is associated with the TaskList instance.
+     */
+    public TaskList(ArrayList<Task> tl, Ui ui) {
+        this.tl = tl;
+        this.ui = ui;
+    }
+
+    /**
+     * serves as the method to return the array list associated
+     * with the TaskList instance.
+     * @return the array list associated with the TaskList instance.
+     */
+    public ArrayList<Task> getArrayList() {
+        return this.tl;
+    }
+
+    /**
+     * Adds task into the array list associated with the TaskList instance.
+     * @param t the task to be added into the array list.
+     */
+    public String add(Task t) {
+        this.tl.add(t);
+        return this.ui.added(t);
+    }
+
+    /**
+     * Adds the task read by Storage class into the array list
+     * associated with this instance of TaskList.
+     * @param ln the line read by the Storage class.
+     */
+    public void add(String ln) {
+        String tag = String.valueOf(ln.charAt(1));
+        String desc = null;
+        int id = -1;
+        boolean done = false;
+        Task toAdd = null;
+        if (String.valueOf(ln.charAt(4)).equals("X")) {
+            done = true;
+        }
+        if (tag.equals("T")) {
```

Comment:
Perhaps this parsing of the command to determine how to create the Todo should be done in the Parser class?